docs/side_quests/nf-test.md

```diff
@@ -0,0 +1,832 @@
+
+    test("Should run without failures") {
+
+        when {
+            params {
+                // define parameters here. Example:
+                // outdir = "tests/results"
+            }
+            process {
+                """
+                // define inputs of the process here. Example:
+                // input[0] = file("test-file.txt")
+                """
+            }
+        }
+
+        then {
+            assert process.success
+            assert snapshot(process.out).match()
+        }
+
+    }
+
+}
+```
+
+This is a similar test to the `sayHello` process, but it's testing the `convertToUpper` process. We know this one will fail because just like with `sayHello`, the `convertToUpper` process takes a single path input, but we haven't specified one.
+
+We now need to supply a single input file to the convertToUpper process, which includes some text that we want to convert to uppercase. There are lots of ways we could do this:
+
+- We could create a dedicated file to test
+- We could re-use the existing data/greetings.csv file
+- We could create it on the fly within the test
+
+For now, let's re-use the existing data/greetings.csv file using the example we used with the pipeline level test.
+
+**Before:**
+
+```groovy title="tests/main.converttoupper.nf.test"
+process {
+    """
+    // define inputs of the process here. Example:
+    // input[0] = file("test-file.txt")
+    """
+}
+```
+
+**After:**
+
+```groovy title="tests/main.converttoupper.nf.test"
+process {
+    """
+    input[0] = "${projectDir}/greetings.csv"
+    """
+}
+```
+
+And run the test!
+
+```bash title="nf-test pipeline pass"
+nf-test test tests/main.converttoupper.nf.test
+```
+
+```console title="nf-test process convertToUpper pass"
+> nf-test test tests/main.converttoupper.nf.test
+
+🚀 nf-test 0.9.2
+https://www.nf-test.com
+(c) 2021 - 2024 Lukas Forer and Sebastian Schoenherr
+
+
+Test Process convertToUpper
+
+  Test [c59b6044] 'Should run without failures' PASSED (1.755s)
+  Snapshots:
+    1 created [Should run without failures]
+
+
+Snapshot Summary:
+  1 created
+
+SUCCESS: Executed 1 tests in 1.764s
+```
+
+Note, we have created a snapshot file for the `convertToUpper` process at `tests/main.converttoupper.nf.test.snap`. If we run the test again, we should see the nf-test passes again.
+
+```bash title="nf-test process convertToUpper pass"
+nf-test test tests/main.converttoupper.nf.test
+```
+
+```console title="nf-test process convertToUpper pass"
+> nf-test test tests/main.converttoupper.nf.test
+
+🚀 nf-test 0.9.2
+https://www.nf-test.com
+(c) 2021 - 2024 Lukas Forer and Sebastian Schoenherr
+
+
+Test Process convertToUpper
+
+  Test [c59b6044] 'Should run without failures' PASSED (1.798s)
+
+
+SUCCESS: Executed 1 tests in 1.811s
+```
+
+### Takeaway
+
+You know how to write tests for a Nextflow process and run them.
+
+### What's next?
+
+Learn how to run tests for everything at once!
+
+## 3. Run tests for the entire repo
```

Comment:
Please write repository in full rather than just repo, as that makes it more understandable for non-English speakers